fileUtils.js

```diff
@@ -0,0 +1,69 @@
+const fs = require('fs');
+const lockMap = new Map();
+
+const writeFile = (path, src) => {
+    return new Promise(async (resolve, reject) => {
+        try {
+            if (lockMap.get(key) && !lockMap.get(key).locked) {
+                const fileHandle = await fs.promises.open(`${process.cwd()}/chunkSrc/${path}`, 'w+');
+                if (fileHandle) {
+                    await fs.promises.writeFile(fileHandle, src);
+                    resolve();
+                }
+            } else {
+                console.error('request for writeFile is touching a locked file, denied.')
+                reject()
+            }
+        } catch (e) {
+            console.error(e);
+            reject();
+        }
+    })
+}
+
+const getAllKeys = async () => {
+    const keys = await fs.promises.readdir('./bin');
+    return keys.length > 0 ? keys : [];
+}
+
+const getFile = (key) => {
+    return new Promise(async (resolve, reject) => {
+        try {
+            if (lockMap.get(key) && !lockMap.get(key).locked) {
+                const fileHandle = await fs.promises.open(`${process.cwd()}/chunkSrc/${key}`, 'r');
+                if (fileHandle) {
+                    const buffer = await fs.promises.readFile(fileHandle);
+                    if (buffer) {
+                        resolve(buffer.toString());
```

Comment:
Why toString? If this is binary data this is not legal.

Reply:
04f45bd